### DISCUSSION TITLE
We have forked October CMS

### DISCUSSION THREAD

**Discussion by LukeTowers** [wintercms/winter#606] [Announcements]
>**UPDATE:** [October CMS has moved to become a locked down paid platform](https://octobercms.com/blog/post/october-cms-moves-become-paid-platform) and is no longer free or open source. Winter CMS will **always** be a free, open source, community driven content management framework.

Former October CMS maintainers [Luke Towers](https://github.com/luketowers), [Ben Thomson](https://github.com/bennothommo) and [Marc Jauvin](https://github.com/mjauvin) wish to announce the formation of the [Winter CMS](https://github.com/wintercms) project, a community-driven fork of [October CMS](https://github.com/octobercms) with a dedication to speed, security, stability and simplicity. Along with [Jack Wilkinson](https://github.com/jaxwilko), we aim to continue to deliver a professional and feature-rich platform that you can rely upon for your websites and applications; as well as engage the community and become a more community-driven project overall.

As indicated though, this is a fork of October CMS. Though it was a regretful decision, the team were forced to do this after a systemic breakdown in communications over a long period of time between the founders of October CMS and the maintenance team.

Winter CMS will continue down the path of continuous improvement while maintaining a solid base that you can rely on to power your business. We aim to maintain complete compatibility with all of your existing October CMS projects, and as much of the future of October CMS as possible within our goals of stability, speed, security, and simplicity. Winter CMS & October CMS are interchangeable as of v1.1.2 & v1.0.472; and we will continue to put in the work to make your life as a developer as simple as possible by minimizing any breaking changes.

## So, what happened?

Between March 3rd-4th 2021, the maintenance team either resigned or were let go from the October CMS project, at the behest of the founders of October CMS.

## Why?

This fork is a result of a distinct lack of communication between the founders and the maintenance team that has formed in the past 2 years, mainly from the founders' side, as well as a general lack of engagement by the founders.

Plans and updates that had apparently been in motion for months were thrust upon the maintenance team without warning at the same time the user-base learned about it. Given that said plans and updates could completely change the scope of maintenance for the project, the maintainer team sought some level of information as to what these plans and updates would entail so they could adequately prepare to maintain the project going forward. These requests were denied.

The vast majority of maintenance work, PR review and implementation, issue management and feature development in the last 2-3 years has been because of the maintenance team as well as the community. Unfortunately, during this time, one of the founders of the project barely communicated or contributed to the project. The other, whilst maintaining the marketplace and marketing, did not interact with the community (except for some marketing) nor the maintainer team at all.

<img width="920" alt="Screen Shot 2021-03-11 at 8 33 23 PM" src="https://user-images.githubusercontent.com/7253840/110885951-61ff3380-82ad-11eb-864d-52864fc9c31d.png">

When it was evident that the founders were unwilling to communicate or collaborate on the project with the maintainers, some of the maintainers decided to leave the October CMS project. Others were then forced out.

## Hasn't this happened before?

Yes it did, in late 2019. Communications and activity from the founders, even at this point, were sorely lacking. Frustrated by this, and a perceived lack of appreciation for the work done by the maintainers from the founders, the maintainers announced their intention to fork. This was, after a day, resolved amicably with the promise of more transparency, hence the maintainer team stayed. See https://github.com/octobercms/october/commit/3e83fbab71adfcf0d4cd3905261faa3ed9b7ff27#commitcomment-47920921 for Sam's point of view on how that went down, and https://shopaholic.one/blog/alexey-bobkov-present-and-future-october-cms for Alexey's point of view on the overall situation.

At the time of that fork, the name October CMF was chosen to minimize the disruption and work required to switch between the projects, mostly in the hopes that reconciliation could occur and the project could continue under a single banner. However, this time around in recognition that there are irreconcilable differences between how the maintainer team and the founders view open source projects; we have chosen to completely rebrand under the name Winter CMS.

## So, what now?

The founders have made it clear that their intention is to proceed with their plans and handle the maintenance of October CMS themselves. We wish them and their users all the best with this endeavour.

The maintainers have therefore decided that to ensure the stability of the project remains intact, we must fork and rebrand as Winter CMS, forming a new team with a clear vision to improve and iterate the platform in less-destructive ways and in a more transparent fashion.

## What does this mean for me?

At this point in time, nothing. You may continue to develop your current projects as before. Plugins and themes developed for October CMS will continue to work in Winter CMS.

The code-bases will only likely diverge with the next release of each product. While the next release of Winter CMS is more iterative, releasing with PHP 8 and Composer 2 support, among several bug fixes and tweaks, it appears that October CMS is intending to release a substantial update which may have an effect on your workflows and plugins. There have already been breaking changes made to the October CMS core (https://github.com/octobercms/october/commit/de897f97482119cafaf1bb94626c2891abd2dc46, https://github.com/octobercms/docs/commit/8e4b38b68c136e73129cd22cef7adf2b0b93b841, https://github.com/octobercms/docs/commit/8ff67e56c737658177078a9d34315cdd159f3a12, https://github.com/octobercms/docs/commit/971177d29e19730fca58830b6c21da4b26119d1c, https://github.com/octobercms/docs/commit/42d7837d7d755e5eeeb1db644e60b351adbac46e#diff-9900aad8c18afff1066263f6a984dff4ff04be29c1d2791398f39862143e4232R11, https://github.com/octobercms/october/issues/5512#issuecomment-799176871 & https://github.com/octobercms/docs/commit/42d7837d7d755e5eeeb1db644e60b351adbac46e#diff-f1c8ca6944983353403f97e69ce7367dc371beb9532d67f84b05f4103f0edd5cR90) and moves toward a more “source-available” mentality than “free and open source” (https://github.com/octobercms/docs/commit/06a3bb37e116f1d5d61b58c1525f7bf15ac9a9e0).

Therefore, if you wish to consider your options before proceeding further, **we strongly recommend you do not update October CMS past version 1.0.472 or version 1.1.2**. You can disable core updates through changing the `disableCoreUpdates` option in `config/cms.php` to `true`, or may lock your Composer dependencies to either of these versions in your `composer.json` file. You may, however, update plugins as you wish.

## You mentioned plugins. What about my plugins?

For Composer plugins, this will not have any effect right now. 

For marketplace plugins, while you are still able to access the full marketplace catalogue as well as any paid plugins you may have access to; there is no guarantee that this will continue to work 100% as the October CMS marketplace is completely under the control of the founders. We are hard at work on a replacement marketplace however, and we encourage plugin developers to submit their plugins to this marketplace if they wish to support Winter CMS. We will have more news on this as development continues.

If you want your plugin to remain compatible with either system while considering your options, you may simply lock updates for that plugin through the Plugin & Updates screen, or you may lock your Composer dependencies to the version of the plugin you are currently using.

## What about my themes?

Themes are in the same boat as plugins if you have purchased them from the marketplace.

## When is Winter CMS releasing?

Winter CMS is available now! See the [guide on our website](https://wintercms.com/blog/post/how-start-using-winter-cms) for how to switch to using it.

## How do I switch to Winter CMS?

See the [guide on our website](https://wintercms.com/blog/post/how-start-using-winter-cms)

## How can I keep up to date with the project?

Feel free to join us on Discord, follow us on Twitter, and sign up for our mailing list on our website, https://wintercms.com. The founders have kicked the maintainers out of all community forums that they have admin access to; so we will be engaging with the community via GitHub, the Discord, Twitter, our mailing list, and our website.

Please also feel free to star our repositories on GitHub, that helps us out a lot!

## Future plans

Our immediate goal is to finish forking the core RainLab plugins and implementing some quality of life improvements (fixing support for the winter:version command, automated splitting of the module repos for faster access to the latest code); as well as fixing some long standing bugs in the platform. We will then be turning our attention towards Laravel 9 LTS compatibility.

If there are any pending PRs / issues on the OctoberCMS repos that you would like dealt with, please resubmit them here. Moving forwards we will continue to work on new builds as per normal and we will also be working on the main website for the project with the goal of eventually replacing the October CMS marketplace and enhancing it beyond its current feature set. In the meantime, you will still be able to use the October CMS marketplace with Winter CMS.

Our big picture goal is to continue to grow and improve and eventually take over WordPress as the CMS of choice around the world. We would like to thank everyone for their patience and understanding as we move forwards and we look forward to all the excellent work that is to come!

Some of the items on the roadmap:
- [x] [Forking RainLab plugins](https://wintercms.com/blog/post/all-rainlab-plugins-have-been-forked)
- [x] Laravel 9 LTS support
- [x] Tailwind support in the Asset Combiner (built in PHP asset compiler) - Replaced by Laravel Mix support in the core
- [x] Backend skin based on Tailwind UI

**Comment by LukeTowers** [comment]
# FAQs

See our blog post for the upgrade guide: https://wintercms.com/blog/post/how-start-using-winter-cms

**Comment by ahangarha** [comment]
Would this project move to some Copy Left license? Otherwise projects like October can exploit WinterCMS and benefit one-sided from community contribution to WinterCMS in their project.

**Comment by ahangarha** [comment]
I am willing to contribute for **Backend skin based on Tailwind UI**. Please let me know where to start and what to keep in mind.

**Comment by bennothommo** [comment]
This project was founded on the MIT license and strong open-source principles, so I don't see the license changing anytime soon. Thanks for the suggestion though.

We do have a Tailwind UI skin for the Backend in the roadmap. When we have more info, we'll post it up, along with any instructions for people who wish to contribute.

**Comment by izdrail** [comment]
I'm glad that somebody is taking this and still keep's it open source.
So I guess If I pull down my projects and replace all the octobercms entries from composer with winter cms ones
I should be ok with using it.
Stupid question I've just read the article on the website and I should be ok.
My only concern is the way you are approaching the composer from what I'm seeing I need to use the dev branch ? Is this just for now until a stable release?

**Comment by LukeTowers** [comment]
@lzomedia you don't have to use the dev branch, but personally I recommend doing so. I run the dev branch on all my sites in production. At the very least I'd recommend using it until 1.1.4 is out because there was a minor annoying bug that was fixed in 1.1.4

**Reply by izdrail** [reply]
Thanks for letting me know. For now I've decided to go for a plain laravel 9 installation for my project . I've used version 9 on the Dev branch and didn't have any problems. I will try to open source the package that I'm working on and make it a plugin at some point

**Reply by LukeTowers** [reply]
@lzomedia FYI we've made a massive amount of progress in the past year and we're literally just about to launch v1.2 with full support for Laravel 9 as a stable release (it's been in testing for the past year via the dev-wip/1.2 branch in composer); so you should be able to resume using Winter for your project and still get to use Laravel 9.

**Comment by izdrail** [comment]
the thing is I'm having to sell this product to enterprise lvl and I cannot say to them we will be using dev branches.
Me as a dev , love the project, enterprise lvl I can't recomand this ( sorry ).
I've been watching this project  (october ) from 2016, been involved in the slack community. Still My question are ? is @daftspunk  still be on the team ? If yes @daftspunk why u are still dooing upgrades on RAINLAB stuff ?

**Comment by LukeTowers** [comment]
@daftspunk is not involved with Winter in any way.
